lib/datadog/appsec/api_security/sampler.rb

```diff
@@ -45,7 +45,9 @@ def sample?(request, response)
           return true if @sample_delay_seconds.zero?
           return false if response.status == 404
 
-          key = Zlib.crc32("#{request.request_method}#{RouteExtractor.route_pattern(request)}#{response.status}")
+          route_pattern = RouteExtractor.route_pattern(request).to_s
```

Comment:
I think it's better to always return something instead, otherwise we might have a bug and just not sampling at all.